src/main/java/com/nha/abdm/wrapper/hip/hrp/link/userInitiated/LinkService.java

```diff
@@ -131,24 +155,51 @@ public void onConfirm(ConfirmResponse confirmResponse) {
     List<CareContext> careContexts = requestLogService.getSelectedCareContexts(linkRefNumber);
 
     OnConfirmPatient onConfirmPatient = null;
+    ResponseEntity<RequestStatusResponse> hipResponse=null;
+    try {
+      log.info("Requesting HIP for verify otp in discovery");
+      VerifyOTP verifyOTP=VerifyOTP.builder().authCode(confirmResponse.getConfirmation().getToken())
+              .loginHint("Discovery OTP request")
+              .linkRefNumber(confirmResponse.getConfirmation().getLinkRefNumber()).build();
+      hipResponse=
+              hipClient.fetchResponseFromHIP(verifyOtpPath,verifyOTP );
+      log.info(verifyOtpPath + " : verifyOtp: " + hipResponse.getStatusCode());
+    } catch (Exception e) {
+      log.error(verifyOtpPath + " : verifyOtp -> Error :" + Exceptions.unwrap(e));
+    }
+    OnConfirmRequest onConfirmRequest=null;
     String tokenNumber = confirmResponse.getConfirmation().getToken();
-    if (tokenNumber.equals("123456")) {
+      assert hipResponse != null;
```

Comment:
why are we using assert?

Reply:
is that the assumption that whatever hipResponse is there, we will assume that otp was matched? should we not have a boolean field in the response which actually conveys this message?